### PR TITLE
move web3 to peerDependencies

### PR DESCRIPTION
**Pull request by VGLoic**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -112,7 +112,9 @@
     "nock": "^13.0.11",
     "nyc": "^15.1.0",
     "prettier": "^2.2.1",
-    "wait-for-expect": "^3.0.2",
+    "wait-for-expect": "^3.0.2"
+  },
+  "peerDependencies": {
     "web3": "^1.2.6"
```
Comment (ricardolyn) [inline package.json:118]: it should still be present on the `devDependencies` if is being used directly in the code. as `peerDependencies` is only to provide information when adding this project as dependency to another that is needed that library

Comment (VGLoic) [reply package.json]: Yup just saw that 🤦 I added back the devDependency

We need to figure out which version we want to add to peerDependencies though, web3.js is currently in ^1.5 so I find it a bit hard to limit the peerDependencies to 1.2.6, I put ^1.x for now but I would need to dig a bit

**Comment by VGLoic** [comment]
**Summary:**

The `web3` version has been updated to latest, some snapshot have been updated in consequence.

The peerDependency for `web3` has been added with version `1.x`.

The badge for vulnerabilities has been added to the README

**Review by achraf17 — APPROVED** [review]
LGTM

**Review by ricardolyn — APPROVED** [review]
LGTM

**Pull request merged**